src/utils/filterTodos.tsx

```diff
@@ -0,0 +1,23 @@
+import { Todo } from '../types/Todo';
+export enum FilterParams {
+  ALL = 'all',
+  COMPLETED = 'completed',
+  ACTIVE = 'active',
+}
+
+export const filterTodos = (
+  todos: Todo[],
+  filterparam: FilterParams,
+): Todo[] => {
+  const resultTodos = [...todos];
+
+  if (filterparam === FilterParams.ACTIVE) {
+    return resultTodos.filter(todo => todo.completed !== true);
+  }
+
+  if (filterparam === FilterParams.COMPLETED) {
+    return resultTodos.filter(todo => todo.completed === true);
+  }
+
+  return resultTodos;
```

Comment:
no need to copy + you can jus wrap it all in a single `filter` call
```suggestion
  return todos.filter(todo => {
    if (filterparam === FilterParams.ACTIVE) {
      return !todo.completed
    }
    
    (filterparam === FilterParams.COMPLETED) {
      return todo.completed
    }
    
    return true;
  })
```